DataFormats/TrackerRecHit2D/interface/BaseTrackerRecHit.h

```diff
@@ -26,7 +26,8 @@ class BaseTrackerRecHit : public TrackingRecHit {
 
   BaseTrackerRecHit(const LocalPoint& p, const LocalError& e, GeomDet const& idet, trackerHitRTTI::RTTI rt)
```

Comment:
Isn't it the case that the `idet` parameter is always a derived `TrackerGeomDet`? The dynamic cast will not work if that is not true, since `det()` returns the pointer to `idet`. If `idet` is always a `TrackerGeomDet`, why not declare it as such, and then line 30 simply becomes:
```
LocalError lape = idet.localAlignmentError();
```

Reply:
The callers for this constructor are all working from lists of generic `GeomDet*` pointers that are assumed to be `TrackerGeomDet*`.  If the type of the argument is changed to 'TrackerGeomDet*', then all the callers have to do a cast, so you end up with even more casts.  I started down that road and eventually concluded that making `GeomDet` usage compile-time type safe would require an enormous refactoring effort.

Reply:
Welcome to the party!